cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -109,13 +121,127 @@ public Void call() throws Exception {
     }
   }
 
+  private void cleanFilesWithoutMeta(final long tillTime, final Map<String, NamespaceId> namespacedLogBaseDirMap,
+                                     final SetMultimap<String, Location> parentDirs) throws Exception {
+    LOG.info("Starting deletion of log files older than {} without metadata", tillTime);
+    List<NamespaceMeta> namespaceMetaList = namespaceQueryAdmin.list();
```

Comment:
not sure I understand the question. The output of ``namespaceQueryAdmin.list()`` does not depend on whether the namespace is impersonated. ``namespaceQueryAdmin.list()`` will simply return the namespaces that the current user has access to, no? So yes, they will be in this list.